Cineaste/CoreDataHelper/MovieStorageManager.swift

```diff
@@ -78,54 +82,77 @@ class MovieStorageManager {
 
             storedMovie.watched = watched
             storedMovie.watchedDate = watched ? Date() : nil
-            self.save(completion: completion)
+
+            self.backgroundContext.saveOrRollback(completion: completion)
         }
     }
 
-    func updateMovieItem(with movie: StoredMovie,
+    func updateMovieItem(with objectID: NSManagedObjectID,
                          watched: Bool,
                          completion: ((_ result: Result<Bool>) -> Void)? = nil) {
         backgroundContext.perform {
-            let storedMovie = StoredMovie(context: self.backgroundContext)
-            storedMovie.id = movie.id
-            storedMovie.title = movie.title
-            storedMovie.overview = movie.overview
-
-            storedMovie.posterPath = movie.posterPath
-            storedMovie.poster = movie.poster
+            guard let storedMovie = self.backgroundContext.object(with: objectID) as? StoredMovie else {
+                fatalError("Object could not be casted to StoredMovie")
+            }
 
-            storedMovie.releaseDate = movie.releaseDate
-            storedMovie.runtime = movie.runtime
-            storedMovie.voteAverage = movie.voteAverage
-            storedMovie.voteCount = movie.voteCount
+            if storedMovie.watched != watched {
+                storedMovie.watched = watched
+            }
+            if storedMovie.watchedDate != (watched ? storedMovie.watchedDate ?? Date() : nil) {
+                storedMovie.watchedDate = watched ? storedMovie.watchedDate ?? Date() : nil
+            }
 
-            storedMovie.watched = watched
-            storedMovie.watchedDate = watched ? Date() : nil
-            self.save(completion: completion)
+            self.backgroundContext.saveOrRollback(completion: completion)
         }
     }
 
     func updateMovieItems(with movies: [StoredMovie],
                           completion: ((_ result: Result<Bool>) -> Void)? = nil) {
         backgroundContext.perform {
             for movie in movies {
-                let storedMovie = StoredMovie(context: self.backgroundContext)
-                storedMovie.id = movie.id
-                storedMovie.title = movie.title
-                storedMovie.overview = movie.overview
-
-                storedMovie.posterPath = movie.posterPath
-                storedMovie.poster = movie.poster
-
-                storedMovie.releaseDate = movie.releaseDate
-                storedMovie.runtime = movie.runtime
-                storedMovie.voteAverage = movie.voteAverage
-                storedMovie.voteCount = movie.voteCount
-
-                storedMovie.watched = movie.watched
-                storedMovie.watchedDate = movie.watchedDate
+                guard let storedMovie = self.backgroundContext.object(with: movie.objectID) as? StoredMovie else {
+                    fatalError("Object could not be casted to StoredMovie")
```

Comment:
Maybe we should not crash the app here, but create a new StoredMovie instance when the object is not found? 
Like this: `let storedMovie = StoredMovie(context: self.backgroundContext)`